src/cryptonote_core/miner.cpp

```diff
@@ -408,5 +528,313 @@ namespace cryptonote
+  {
+    if( is_background_mining_enabled && !epee::file_io_utils::is_file_exist("/proc/stat") )
+    {
+      LOG_ERROR("failed to find file /proc/stat, background mining will not be enabled / remain disabled :(");
+      m_is_background_mining_enabled = false;
+      return false;
+    }
+
+    if( m_is_background_mining_enabled == is_background_mining_enabled )
+    {
+      LOG_PRINT_RED_L0("attempt to set m_is_background_mining_enabled to same value ... pointless");
+      return false;
+    }
+
+    m_is_background_mining_enabled = is_background_mining_enabled;
+    m_cpu_data_start.update_time = 0;
+    m_cpu_data_end.update_time = 0;
+
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  uint64_t miner::get_min_idle_seconds() const
+  {
+    return m_min_idle_seconds;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_min_idle_seconds(const uint64_t& min_idle_seconds)
+  {
+    if(min_idle_seconds > BACKGROUND_MINING_MAX_MIN_IDLE_INTERVAL_IN_SECONDS) return false;
+    if(min_idle_seconds < BACKGROUND_MINING_MIN_MIN_IDLE_INTERVAL_IN_SECONDS) return false;
+    m_min_idle_seconds = min_idle_seconds;
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  uint8_t miner::get_idle_threshold() const
+  {
+    return m_idle_threshold;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_idle_threshold(const uint8_t& idle_threshold)
+  {
+    if(idle_threshold > BACKGROUND_MINING_MAX_IDLE_THRESHOLD_PERCENTAGE) return false;
+    if(idle_threshold < BACKGROUND_MINING_MIN_IDLE_THRESHOLD_PERCENTAGE) return false;
+    m_idle_threshold = idle_threshold;
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  uint8_t miner::get_mining_threshold() const
+  {
+    return m_mining_threshold;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_mining_threshold(const uint8_t& mining_threshold)
+  {
+    if(mining_threshold > BACKGROUND_MINING_MAX_MINING_THRESHOLD_PERCENTAGE) return false;
+    if(mining_threshold < BACKGROUND_MINING_MIN_MINING_THRESHOLD_PERCENTAGE) return false;
+    m_mining_threshold = mining_threshold;
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::get_cpu_stats(cpu_data& cpud)
+  {
+    #if defined(WIN32) || defined(_WIN32) || defined(__WIN32)
+
+    LOG_PRINT_RED_L0("background mining not supported on windows ... yet");
+    return false;
+
+    #else
+
+    const std::string STR_CPU("cpu");
```

Comment:
I _think_ this stuff does not work on BSD based systems, either. Should the macro guard check for anything `!Linux`?